packages/protocol/test/ZkAsset/ZkAssetBurnable.js

```diff
@@ -0,0 +1,260 @@
+            const linkedTokenInitialBalance = (await erc20.balanceOf(accounts[0])).toNumber();
+            const aceInitialBalance = (await erc20.balanceOf(ace.address)).toNumber();
+
+            expect(linkedTokenInitialBalance).to.equal(-kPublic);
+            expect(aceInitialBalance).to.equal(0);
+
+            const { receipt: joinSplitReceipt } = await zkAssetBurnable.confidentialTransfer(proofs[0].proofData);
+            expect(joinSplitReceipt.status).to.equal(true);
+
+            const linkedTokenIntermediatateBalance = (await erc20.balanceOf(accounts[0])).toNumber();
+            expect(linkedTokenIntermediatateBalance).to.equal(0);
+            const aceIntermediateBalance = (await erc20.balanceOf(ace.address)).toNumber();
+            expect(aceIntermediateBalance).to.equal(-kPublic);
+
+            const { receipt: burnReceipt } = await zkAssetBurnable.confidentialBurn(BURN_PROOF, proofs[1].proofData);
+            expect(burnReceipt.status).to.equal(true);
+
+            const aceBalanceFinal = (await erc20.balanceOf(ace.address)).toNumber();
+            const linkedTokenFinalBalance = (await erc20.balanceOf(accounts[0])).toNumber();
+
+            expect(aceBalanceFinal).to.equal(-kPublic);
+            expect(linkedTokenFinalBalance).to.equal(0);
+        });
+    });
+
+    describe('failure states', () => {
+        let aztecAccounts = [];
+        let notes = [];
+        let ace;
+        const proofs = [];
+        let erc20;
+        let zkAssetBurnable;
+        let scalingFactor;
+        let aztecAdjustSupply;
+        let aztecJoinSplit;
+        let adjustedNotes;
+        let oldTotalBurned;
+        let zeroNote;
+
+        beforeEach(async () => {
+            ace = await ACE.new({ from: accounts[0] });
+            aztecAdjustSupply = await AdjustSupply.new();
+            aztecJoinSplit = await JoinSplit.new();
+
+            aztecAccounts = [...new Array(4)].map(() => secp256k1.generateAccount());
+            const noteValues = [50, 0, 30, 20]; // note we do not use this third note, we create fixed one
+            notes = aztecAccounts.map(({ publicKey }, i) => {
+                return note.create(publicKey, noteValues[i]);
+            });
+
+            await ace.setCommonReferenceString(constants.CRS);
+            await ace.setProof(BURN_PROOF, aztecAdjustSupply.address);
+            await ace.setProof(JOIN_SPLIT_PROOF, aztecJoinSplit.address);
+
+
+            // Creating a fixed note
+            const a = padLeft('1', 64);
+            const k = padLeft('0', 8);
+            const ephemeral = secp256k1.ec.keyFromPrivate(crypto.randomBytes(32));
+            const viewingKey = `0x${a}${k}${padLeft(ephemeral.getPublic(true, 'hex'), 66)}`;
+            zeroNote = note.fromViewKey(viewingKey);
+
+            erc20 = await ERC20Mintable.new();
+            scalingFactor = new BN(1);
+        });
+
+        it('validates failure if burn attempted when flag set to false', async () => {
```

Comment:
Should add a test here that throws if user that does not own `zkAsset` attempts to burn a note

Reply:
Moved to testing checklist